app/views/servers/check_installation.js.erb

```diff
@@ -2,5 +2,20 @@
   $("[data-server-status~=checking]").addClass("hidden");
   $("[data-server-status~=connected]").removeClass("hidden");
 <% else %>
+  <% if @server.install_step == 1 %>
+    $(".step1").addClass("info").find("td:first-child").html("<%= j icon "circle-o-notch fa-spin" %>");
+  <% elsif @server.install_step == 2 %>
+    $(".step1").removeClass("info").addClass("success").find("td:first-child").html("<%= j icon "check-square-o" %>");
+    $(".step2").addClass("info").find("td:first-child").html("<%= j icon "circle-o-notch fa-spin" %>");
+  <% elsif @server.install_step == 3 %>
+    $(".step1").removeClass("info").addClass("success").find("td:first-child").html("<%= j icon "check-square-o" %>");
```

Comment:
Ah right I see. Maybe refactor this into passing a `steps_completed` array so you don't have to "hardcode" the step numbers to the messages, but can set "success" classes dynamically by looping over the `steps_completed` that are the same as the classes on the status line. As in:

```
@steps_completed.each do |step_name|
  $('.step').removeClass('info, success')
  $('.step-<%= step_name %>').addClass("success")
end
```